src/test/java/spoon/test/ctElement/CtElementImplTest.java

```diff
@@ -0,0 +1,41 @@
+package spoon.test.ctElement;
+
+import org.junit.jupiter.api.Test;
+import spoon.Launcher;
+import spoon.reflect.declaration.CtAnnotation;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.factory.Factory;
+import spoon.support.compiler.FileSystemFile;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class CtElementImplTest {
+
+    @Test
+    public void testRemoveAnnotation() {
+        // contract: removeAnnotation returns true after removing an annotation of a class containing a single
+        // annotation, and returns false when a non existing annotation is tried to be removed
+
+        // arrange
+        Launcher spoon = new Launcher();
+        spoon.addInputResource(new FileSystemFile("./src/test/java/spoon/test/ctElement/CtElementImplTest.java"));
+        spoon.buildModel();
+        Factory factory = spoon.getFactory();
+
+        CtClass<Annotated> annotatedClass = factory.Class().get(Annotated.class);
+        assertEquals(1, annotatedClass.getAnnotations().size());
+        CtAnnotation annotationToBeRemoved = annotatedClass.getAnnotations().get(0);
+
+        // act
+        Boolean expectedBoolean = annotatedClass.removeAnnotation(annotationToBeRemoved);
+        Boolean expectedBoolean2 = annotatedClass.removeAnnotation(annotationToBeRemoved);
```

Comment:
Small comments,

here, you could use `boolean` instead of `Boolean`. `CtElement#removeAnnotation` returns boolean. This change saves a reader the noise if the Boolean could be ever null.

Renaming `expectedBoolean` to something like `shouldBeTrue` and `expectedBoolean2`
to e.g. `shouldBeFalse` helps a reader and is a way more meaningful name.

Reply:
Yes, you are right, thanks for the wonderful suggestions.

Feel free to contribute suggestions : )

Reply:
Thanks @MartinWitt!

While this naming explains the expected result, it doesn't well represent _what_ the values are. One could for example consider this naming instead:

```java
        boolean firstRemovalSuccessful = annotatedClass.removeAnnotation(annotationToBeRemoved);
        boolean secondRemovalSuccessful = annotatedClass.removeAnnotation(annotationToBeRemoved);
```

This doesn't encode what the expected value is, but that is in fact encoded in the assertion. I'm not putting this in as a review comment as I think the names suggested by @MartinWitt are perfectly serviceable and read just fine, I just wanted to plant the thought.

Reply:
On a side note: if you ever feel the need to enumerate variables (e.g. `something1`, `something2` etc), that's a clear indication that you need to reconsider the naming :)